### PR TITLE
Introduce share-path argument to allow a VM to mount a host directory

### PR DESCRIPTION
**Pull request by marcosps**
I'm adding this new argument in to make it easier to share test script to run inside the guests. I had to change the numa assignment inside the methods that export the virtiofsd, so please take a look if this is correct!

Currently it works for me, but maybe there are corner cases in the code that I didn't test.

### CODE REVIEW

File: virtme/commands/run.py [hunk 1]
```diff
@@ -179,6 +179,13 @@ def make_parser() -> argparse.ArgumentParser:
         metavar="NAME=PATH",
         help="Add a read/write virtio-blk disk.  The device nodes will be /dev/disk/by-id/virtio-virtme_disk_blk_NAME.",
     )
+    g.add_argument(
+        "--shared-path",
+        action="append",
+        default=[],
+        metavar="TAG=PATH",
+        help="Share a direct using virtiofs with the guest.",
```
Comment (matttbe) [inline virtme/commands/run.py]: I think a word is missing here. → _Share a direct XXX using (...)_

**Review by matttbe — CHANGES_REQUESTED** [review]
That would be great to have multiple mount points with virtiofs! Just some suggestions: I initially didn't understand what you wanted to do when looking at the new option and description :)

(also, please squash the commit added by the CI :) )

File: virtme/commands/run.py [hunk 2]
```diff
@@ -1323,6 +1353,26 @@ def do_it() -> int:
...
+        tag, path = sanitize_shared_path_args(arg)
+        virtiofs_config = VirtioFSConfig(
+            path=path,
+            mount_tag=tag,
+            guest_tools_path=guest_tools_path,
+            # virtiofsd requires a NUMA not, if --numa is specified simply use
+            # the user-defined NUMA node, otherwise create a NUMA node with all
+            # the memory.
+            memory=args.memory,
+            rw=args.rw,
```
[the remark this answers is not in the document]
Comment (matttbe) [reply virtme/commands/run.py]: Good point! If it is in RW, we might want to have an option to control `virtiofsd -o cache=...`, see #353

**Comment by marcosps** [comment]
> I guess the benefits of this is having a faster shared dir compared to --rwdir / --rodir, right?
> 
> I haven't tested this yet, but do we need to manually mount the shared dirs in the guest (I was expecting to see some changes in virtme-init/virtme-ng-init as well)?

I forgot to tell that this works when `--root` is passed, so this feature is similar to using the `--volume` on containers, mapping a dir from the host inside the VM.

File: virtme/commands/run.py (hunk 1 above)
Comment (marcosps) [reply virtme/commands/run.py]: Well, sorry for not making it clear. I'm using this new option with `--root`, so the directory is not available to the VM. I'm trying to test different versions of SLE, with different userspace rootfs, but my test scripts and livepatches are on the host, so I'm using virtiofsd to make them available to the guest.

Regarding this usage, I'm not sure if this is similar to what you proposed. I never used `overlay-rwdir` so I;m not how this can be used ...

**Comment by matttbe** [comment]
> @matttbe @arighi since I'm using this new option together with `--root`, maybe we can restrict it to the same situation? I mean, it currently doesn't make much sense to use it when we are sharing the same rootfs with the host, right?

@marcosps I think it can be useful to share some directories in RW, while the rest is in an "overlayfs" mode. (In this case, we might need something similar to #355)

So having the same FS as the host, except some directories that are "shared" to pass data, but having a better performance than using 9P. No?

File: virtme/commands/run.py (hunk 1 above)
Comment (matttbe) [reply virtme/commands/run.py]: OK but then `--rodir` might be enough for your case then? Or do you need something faster? (I think it still makes sense to have it, but maybe you don't really need it for your case?)

**Comment by marcosps** [comment]
> > @matttbe @arighi since I'm using this new option together with `--root`, maybe we can restrict it to the same situation? I mean, it currently doesn't make much sense to use it when we are sharing the same rootfs with the host, right?
> 
> @marcosps I think it can be useful to share some directories in RW, while the rest is in an "overlayfs" mode. (In this case, we might need something similar to #355)
> 
> So having the same FS as the host, except some directories that are "shared" to pass data, but having a better performance than using 9P. No?

Does `--overlay-rw` woprk with rootfs? I've being trying to read the code and checking using the flag (it doesn't really describe how it should be used, but per the code it seems that it needs a /guest/path=/home/path argument, like --sahred-path).

But I can't see the the guest directory being populated by vng. Am I using it wrong?

**Comment by marcosps** [comment]
(Sorry to flood here...)
@matttbe ah, ok, now I get what you and @arighi were talking about: `--rodir` can also map directories to the guest, but for now I can't do that when using `--root` with /mnt or /tmp for example:

```
error: cannot initialize /tmp/ inside the guest (path must be defined inside a valid overlay)
```

I'll check what it means...
